src/backend/distributed/commands/multi_copy.c

```diff
@@ -974,7 +850,7 @@ OpenCopyTransactions(CopyStmt *copyStatement, ShardConnections *shardConnections
 
 	MemoryContextSwitchTo(oldContext);
 
-	if (XactModificationLevel > XACT_MODIFICATION_NONE)
+	if (XactModificationLevel > XACT_MODIFICATION_DATA)
```

Comment:
Since we're enabling DML + COPY, we should expand our test coverage, e.g. add some tests in which we insert into multiple different shards, such that COPY becomes impossible.